src/systemd/sd-json.h

```diff
@@ -0,0 +1,334 @@
+  - Formatter has color, line, column support
+
+  Limitations:
+  - Doesn't allow embedded NUL in strings
+  - Can't store integers outside of the -9223372036854775808…18446744073709551615 range (it will use 'double' for
+    values outside this range, which is lossy)
+  - Can't store negative zero (will be treated identical to positive zero, and not retained across serialization)
+  - Can't store non-integer numbers that can't be stored in "double" losslessly
+  - Allows creation and parsing of objects with duplicate keys. The "dispatcher" will refuse them however. This means
+    we can parse and pass around such objects, but will carefully refuse them when we convert them into our own data.
+
+  (These limitations should be pretty much in line with those of other JSON implementations, in fact might be less
+  limiting in most cases even.)
+*/
+
+typedef struct sd_json_variant sd_json_variant;
+
+__extension__ typedef enum sd_json_variant_type_t {
+        SD_JSON_VARIANT_STRING,
+        SD_JSON_VARIANT_INTEGER,
+        SD_JSON_VARIANT_UNSIGNED,
+        SD_JSON_VARIANT_REAL,
+        SD_JSON_VARIANT_NUMBER, /* This a pseudo-type: we can never create variants of this type, but we use it as wildcard check for the above three types */
+        SD_JSON_VARIANT_BOOLEAN,
+        SD_JSON_VARIANT_ARRAY,
+        SD_JSON_VARIANT_OBJECT,
+        SD_JSON_VARIANT_NULL,
+        _SD_JSON_VARIANT_TYPE_MAX,
+        _SD_JSON_VARIANT_TYPE_INVALID = -EINVAL,
+        _SD_ENUM_FORCE_S64(JSON_VARIANT_TYPE)
+} sd_json_variant_type_t;
+
+int sd_json_variant_new_string(sd_json_variant **ret, const char *s);
+int sd_json_variant_new_stringn(sd_json_variant **ret, const char *s, size_t n);
+int sd_json_variant_new_base64(sd_json_variant **ret, const void *p, size_t n);
+int sd_json_variant_new_base32hex(sd_json_variant **ret, const void *p, size_t n);
+int sd_json_variant_new_hex(sd_json_variant **ret, const void *p, size_t n);
+int sd_json_variant_new_octescape(sd_json_variant **ret, const void *p, size_t n);
+int sd_json_variant_new_integer(sd_json_variant **ret, int64_t i);
+int sd_json_variant_new_unsigned(sd_json_variant **ret, uint64_t u);
+int sd_json_variant_new_real(sd_json_variant **ret, double d);
+int sd_json_variant_new_boolean(sd_json_variant **ret, int b);
+int sd_json_variant_new_array(sd_json_variant **ret, sd_json_variant **array, size_t n);
+int sd_json_variant_new_array_bytes(sd_json_variant **ret, const void *p, size_t n);
+int sd_json_variant_new_array_strv(sd_json_variant **ret, char **l);
+int sd_json_variant_new_object(sd_json_variant **ret, sd_json_variant **array, size_t n);
+int sd_json_variant_new_null(sd_json_variant **ret);
+int sd_json_variant_new_id128(sd_json_variant **ret, sd_id128_t id);
+int sd_json_variant_new_uuid(sd_json_variant **ret, sd_id128_t id);
+
+sd_json_variant *sd_json_variant_ref(sd_json_variant *v);
+sd_json_variant *sd_json_variant_unref(sd_json_variant *v);
```

Comment:
Nowadays, we use e.g. `sd_json_variant* sd_json_variant_ref()`, that is, space after `*` rather than before.